src/main/groovy/com/netflix/nebula/lint/rule/dependency/provider/ClasspathBasedRecommendationProvider.java

```diff
@@ -62,10 +57,8 @@ Set<File> getBomsOnConfiguration() {
                                 .create(dep.getGroup() + ":" + dep.getName() + ":" + dep.getVersion() + "@pom"));
 
                 final Set<File> files = new LinkedHashSet<>();
-                // TODO: Replace this following  https://docs.gradle.org/8.8-rc-1/userguide/upgrading_version_8.html#deprecate_filtered_configuration_file_and_filecollection_methods
-                DeprecationLogger.whileDisabled(() -> {
-                    files.addAll(bomConf.getResolvedConfiguration().getLenientConfiguration().getFiles(Specs.SATISFIES_ALL));
-                });
+                files.addAll(bomConf.getResolvedConfiguration().getLenientConfiguration().getArtifacts().stream().map(ResolvedArtifact::getFile).collect(Collectors.toSet()));
```

Comment:
I think this provides the same end-result... also, we run this after resolution so it should be fine